src/exporters/datadog/mod.rs

```diff
@@ -2,46 +2,46 @@
 
 use crate::bounded_channel::BoundedReceiver;
 use crate::exporters::datadog::request_builder::RequestBuilder;
-use crate::exporters::datadog::request_builder_mapper::RequestBuilderMapper;
 use crate::exporters::datadog::transform::Transformer;
 use crate::exporters::http;
 use crate::exporters::http::retry::{RetryConfig, RetryPolicy};
 
 use crate::exporters::http::client::ResponseDecode;
+use crate::exporters::http::exporter::{Exporter, ResultLogger};
 use crate::exporters::http::http_client::HttpClient;
+use crate::exporters::http::request_builder_mapper::RequestBuilderMapper;
 use crate::exporters::http::response::Response;
 use crate::exporters::http::types::ContentEncoding;
-use crate::topology::flush_control::{FlushReceiver, conditional_flush};
+use crate::topology::flush_control::FlushReceiver;
 use bytes::Bytes;
 use flume::r#async::RecvStream;
-use futures::stream::FuturesUnordered;
 use http_body_util::Full;
 use opentelemetry_proto::tonic::trace::v1::ResourceSpans;
-use std::error::Error;
-use std::future::Future;
-use std::ops::Add;
-use std::pin::Pin;
 use std::time::Duration;
-use tokio::select;
-use tokio::time::{Instant, timeout_at};
-use tokio_stream::StreamExt;
-use tokio_util::sync::CancellationToken;
 use tower::retry::Retry as TowerRetry;
 use tower::timeout::Timeout;
-use tower::{BoxError, Service, ServiceBuilder};
-use tracing::log::warn;
-use tracing::{Level, debug, error, event};
+use tower::{BoxError, ServiceBuilder};
+use tracing::error;
 
 mod api_request;
 mod request_builder;
-mod request_builder_mapper;
 mod transform;
 mod types;
 
-const MAX_CONCURRENT_REQUESTS: usize = 10;
-
-type ExportFuture =
-    Pin<Box<dyn Future<Output = Result<Response<()>, Box<dyn Error + Send + Sync>>> + Send>>;
+type SvcType =
+    TowerRetry<RetryPolicy<()>, Timeout<HttpClient<Full<Bytes>, (), DatadogTraceDecoder>>>;
+
+type ExporterType<'a, Resource> = Exporter<
```

Comment:
How did we end up needing lifetimes in this PR?

Reply:
The lifetime comes from the input stream, the bounded channel. When we convert it into a stream it has this lifetime attached: https://github.com/streamfold/rotel/blob/7674718d32cef611c815997423ef702b50adcaa0/src/bounded_channel.rs#L63